backend/src/test/java/com/crm/service/impl/AccountServiceImplTest.java

```diff
@@ -0,0 +1,338 @@
+import com.crm.util.EmailUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest
+@RunWith(MockitoJUnitRunner.class)
+public class AccountServiceImplTest {
+
+    @Mock
+    private AccountRepository accountRepository;
+
+    @Mock
+    private RoleRepository roleRepository;
+
+    @Mock
+    private AccountMapper accountMapper;
+
+    @Mock
+    private EmailUtils emailUtils;
+
+    @Mock
+    private VerificationTokenServiceImpl verificationTokenService;
+
+    @Spy
+    private PasswordEncoder passwordEncoder;
+
+    @InjectMocks
+    private AccountServiceImpl accountService;
+
+    private AccountEntity accountEntity;
+    private AccountEntity savedAccountEntity;
+    private AccountEntity activatedAccountEntity;
+    private AccountResponse accountResponse;
+    private NewAccountRequest newAccountRequest;
+    private VerificationTokenEntity neverExpiredVerificationToken;
+    private VerificationTokenEntity expiredVerificationToken;
+
+
+    private final Long id = 10L;
+    private final String login = "Test485";
+    private final String email = "Test485";
+    private final String password = "Test485";
+    private final boolean notActivated = false;
+    private final boolean activated = true;
+    private final Timestamp lastFailedLogin = Timestamp.from(Instant.now());
+    private final Timestamp registrationDate = Timestamp.from(Instant.now());
+    private final Timestamp lastSuccessfulLogin = Timestamp.from(Instant.now());
+    private final int loginAttempts = 1;
+    private final RoleEntity role = RoleEntity.builder().id(1L).name("admin").build();
+    private final String verificationToken = "tokenToken";
+    private final LocalDateTime neverExpiredToken = LocalDateTime.of(2050, 1, 1, 1, 1, 1);
+    private final LocalDateTime expiredToken = LocalDateTime.of(2010, 1, 1, 1, 1, 1);
+    private final String userRole = "ROLE_USER";
+
+
+    @Before
```

Comment:
I would use @BeforeClass or @BeforeAll to generate objects only once, not before every test (@Before is from Junit4 and runs before each test)